src/transaction_builder.cpp

```diff
@@ -67,16 +89,15 @@ void TransactionBuilder::AddTransparentInput(COutPoint utxo, CScript scriptPubKe
     tIns.emplace_back(scriptPubKey, value);
 }
 
-bool TransactionBuilder::AddTransparentOutput(CTxDestination& to, CAmount value)
+void TransactionBuilder::AddTransparentOutput(CTxDestination& to, CAmount value)
 {
     if (!IsValidDestination(to)) {
-        return false;
+        throw std::runtime_error("Invalid output address, not a valid taddr.");
```

Comment:
Do the calling functions need an explicit bool back? It seems odd we could just get away with returning void here

Reply:
Similar to the above, either this should not happen (when adding an already validated change address) or we were using it as follows:
```
if (!builder_.AddTransparentOutput(address, amount)) {
    throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid output address, not a valid taddr.");
}
```
In which case the user will see a similar error.